frostsnapp/native/src/wallet.rs

```diff
@@ -181,22 +192,48 @@ impl _Wallet {
         })
     }
 
-    fn get_descriptor(frost_key: &FrostKey<Normal>) -> Descriptor<DescriptorPublicKey> {
-        use bitcoin::bip32::DerivationPath;
-        let frost_xpub = frostsnap_core::xpub::Xpub::new(frost_key.clone());
-        let key = DescriptorPublicKey::XPub(DescriptorXKey {
-            origin: None,
-            xkey: *frost_xpub.xpub(),
-            derivation_path: DerivationPath::master(),
-            wildcard: Wildcard::Unhardened,
-        });
+    fn get_descriptors(
+        frost_key: &FrostKey<Normal>,
+    ) -> Vec<(Keychain, Descriptor<DescriptorPublicKey>)> {
+        let secp = secp256k1::Secp256k1::verification_only();
 
-        let tr = Tr::new(key, None).expect("infallible since it's None");
-        Descriptor::Tr(tr)
+        let (app_key, chaincode) =
+            frost_key.app_tweak_and_expand(frostsnap_core::tweak::AppTweakKind::Bitcoin);
+        let root_bitcoin_xpub = *frostsnap_core::tweak::Xpub::new(app_key, chaincode).xpub();
+
+        [Keychain::External, Keychain::Internal]
+            .into_iter()
+            .map(move |keychain| {
```

Comment:
I'm not sure this requires `move` since we're already taking ownership through `into_iter` and nothing in the closure is getting mutated?